Fix compilation using exiv2-0.27

Compilation is currently broken on the configuration stage when using exiv2-0.27.

Patch from @darix 
https://build.opensuse.org/package/rdiff/home:darix:playground/hdrmerge?linkrev=base&rev=5

I'd like to add that I get the following error. tho I'm not sure it's related
```
[ 53%] Building CXX object CMakeFiles/hdrmerge.dir/src/Launcher.cpp.o
/Users/imad/hdrmerge/src/main.cpp:23:10: fatal error: 'exiv2/error.hpp' file not found
#include <exiv2/error.hpp>
         ^~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/hdrmerge.dir/src/main.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/hdrmerge.dir/all] Error 2
make: *** [all] Error 2
```

@darix @Beep6581 On system MacOS 10.14.5ß18F118d, `cmake-exiv2` branch, latest cmake, this is what I get:


```
-- Found PkgConfig: /opt/local/bin/pkg-config (found version "0.29.2") 
-- Checking for module 'libraw'
--   Found libraw, version 0.17.2
-- Checking for module 'libraw_r'
--   Found libraw_r, version 0.17.2
-- Checking for module 'exiv2'
--   Found exiv2, version 0.27.1.1
-- Found Exiv2: /opt/local/lib/libexiv2.dylib (Required is at least version "0.21") 
```
...
```
[100%] Linking CXX executable hdrmerge.app/Contents/MacOS/hdrmerge
/opt/local/bin/cmake -E cmake_link_script CMakeFiles/hdrmerge.dir/link.txt --verbose=1
/usr/bin/clang++   -std=c++11 -Wall -Wno-unknown-pragmas -Xpreprocessor -fopenmp /opt/local/lib/libiomp5.dylib -I/opt/local/include -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/hdrmerge.dir/hdrmerge_autogen/mocs_compilation.cpp.o CMakeFiles/hdrmerge.dir/src/main.cpp.o CMakeFiles/hdrmerge.dir/src/Launcher.cpp.o CMakeFiles/hdrmerge.dir/src/Image.cpp.o CMakeFiles/hdrmerge.dir/src/ImageStack.cpp.o CMakeFiles/hdrmerge.dir/src/Bitmap.cpp.o CMakeFiles/hdrmerge.dir/src/RawParameters.cpp.o CMakeFiles/hdrmerge.dir/src/EditableMask.cpp.o CMakeFiles/hdrmerge.dir/src/DngFloatWriter.cpp.o CMakeFiles/hdrmerge.dir/src/TiffDirectory.cpp.o CMakeFiles/hdrmerge.dir/src/BoxBlur.cpp.o CMakeFiles/hdrmerge.dir/src/ExifTransfer.cpp.o CMakeFiles/hdrmerge.dir/src/ImageIO.cpp.o CMakeFiles/hdrmerge.dir/src/AboutDialog.cpp.o CMakeFiles/hdrmerge.dir/src/MainWindow.cpp.o CMakeFiles/hdrmerge.dir/src/PreviewWidget.cpp.o CMakeFiles/hdrmerge.dir/src/DraggableScrollArea.cpp.o CMakeFiles/hdrmerge.dir/src/DngPropertiesDialog.cpp.o CMakeFiles/hdrmerge.dir/src/LoadOptionsDialog.cpp.o CMakeFiles/hdrmerge.dir/src/FileSystem.cpp.o CMakeFiles/hdrmerge.dir/hdrmerge_autogen/QYFM2Z2WYQ/qrc_resources.cpp.o CMakeFiles/alglib-objects.dir/alglib-objects_autogen/mocs_compilation.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/alglibinternal.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/alglibmisc.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/ap.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/dataanalysis.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/diffequations.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/fasttransforms.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/integration.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/interpolation.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/linalg.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/optimization.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/solvers.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/specialfunctions.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/statistics.cpp.o  -o hdrmerge.app/Contents/MacOS/hdrmerge -Wl,-rpath,/Users/rb/Qt596/5.9.6/clang_64/lib -lraw_r -lstdc++ -llcms2 /opt/local/lib/libz.dylib /Users/rb/Qt596/5.9.6/clang_64/lib/QtWidgets.framework/QtWidgets -lexiv2 /Users/rb/Qt596/5.9.6/clang_64/lib/QtGui.framework/QtGui /Users/rb/Qt596/5.9.6/clang_64/lib/QtCore.framework/QtCore 
ld: library not found for -lraw_r
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [hdrmerge.app/Contents/MacOS/hdrmerge] Error 1
make[1]: *** [CMakeFiles/hdrmerge.dir/all] Error 2
make: *** [all] Error 2
```

```
$  ls -la /opt/local/lib/libraw*
-rwxr-xr-x  1 root  wheel  1141440 Apr 25 08:38 /opt/local/lib/libraw.15.dylib
-rw-r--r--  1 root  wheel  1284232 Apr 25 08:38 /opt/local/lib/libraw.a
lrwxr-xr-x  1 root  wheel       15 Apr 25 08:38 /opt/local/lib/libraw.dylib -> libraw.15.dylib
-rwxr-xr-x  1 root  wheel      998 Apr 25 08:38 /opt/local/lib/libraw.la
-rwxr-xr-x  1 root  wheel  1141440 Apr 25 08:38 /opt/local/lib/libraw_r.15.dylib
-rw-r--r--  1 root  wheel  1284232 Apr 25 08:38 /opt/local/lib/libraw_r.a
lrwxr-xr-x  1 root  wheel       17 Apr 25 08:38 /opt/local/lib/libraw_r.dylib -> libraw_r.15.dylib
-rwxr-xr-x  1 root  wheel     1010 Apr 25 08:38 /opt/local/lib/libraw_r.la
```

```
$ cat /opt/local/lib/pkgconfig/libraw_r.pc
prefix=/opt/local
exec_prefix=${prefix}
libdir=/opt/local/lib
includedir=${prefix}/include

Name: libraw
Description: Raw image decoder library (thread-safe)
Requires: lcms2
Version: 0.17.2
Libs: -L${libdir} -lraw_r -lstdc++
Cflags: -I${includedir}/libraw
```

Also, exiv2, a similar prob.

```
[100%] Linking CXX executable hdrmerge.app/Contents/MacOS/hdrmerge
/opt/local/bin/cmake -E cmake_link_script CMakeFiles/hdrmerge.dir/link.txt --verbose=1
/usr/bin/clang++   -std=c++11 -Wall -Wno-unknown-pragmas -Xpreprocessor -fopenmp /opt/local/lib/libiomp5.dylib -I/opt/local/include -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/hdrmerge.dir/hdrmerge_autogen/mocs_compilation.cpp.o CMakeFiles/hdrmerge.dir/src/main.cpp.o CMakeFiles/hdrmerge.dir/src/Launcher.cpp.o CMakeFiles/hdrmerge.dir/src/Image.cpp.o CMakeFiles/hdrmerge.dir/src/ImageStack.cpp.o CMakeFiles/hdrmerge.dir/src/Bitmap.cpp.o CMakeFiles/hdrmerge.dir/src/RawParameters.cpp.o CMakeFiles/hdrmerge.dir/src/EditableMask.cpp.o CMakeFiles/hdrmerge.dir/src/DngFloatWriter.cpp.o CMakeFiles/hdrmerge.dir/src/TiffDirectory.cpp.o CMakeFiles/hdrmerge.dir/src/BoxBlur.cpp.o CMakeFiles/hdrmerge.dir/src/ExifTransfer.cpp.o CMakeFiles/hdrmerge.dir/src/ImageIO.cpp.o CMakeFiles/hdrmerge.dir/src/AboutDialog.cpp.o CMakeFiles/hdrmerge.dir/src/MainWindow.cpp.o CMakeFiles/hdrmerge.dir/src/PreviewWidget.cpp.o CMakeFiles/hdrmerge.dir/src/DraggableScrollArea.cpp.o CMakeFiles/hdrmerge.dir/src/DngPropertiesDialog.cpp.o CMakeFiles/hdrmerge.dir/src/LoadOptionsDialog.cpp.o CMakeFiles/hdrmerge.dir/src/FileSystem.cpp.o CMakeFiles/hdrmerge.dir/hdrmerge_autogen/QYFM2Z2WYQ/qrc_resources.cpp.o CMakeFiles/alglib-objects.dir/alglib-objects_autogen/mocs_compilation.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/alglibinternal.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/alglibmisc.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/ap.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/dataanalysis.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/diffequations.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/fasttransforms.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/integration.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/interpolation.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/linalg.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/optimization.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/solvers.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/specialfunctions.cpp.o CMakeFiles/alglib-objects.dir/Users/rb/alglib/cpp/src/statistics.cpp.o  -o hdrmerge.app/Contents/MacOS/hdrmerge -Wl,-rpath,/Users/rb/Qt596/5.9.6/clang_64/lib /opt/local/lib/libraw_r.dylib /opt/local/lib/libz.dylib /Users/rb/Qt596/5.9.6/clang_64/lib/QtWidgets.framework/QtWidgets -lexiv2 /Users/rb/Qt596/5.9.6/clang_64/lib/QtGui.framework/QtGui /Users/rb/Qt596/5.9.6/clang_64/lib/QtCore.framework/QtCore 
ld: library not found for -lexiv2
```

To enable compilation, I hardcoded the libs:
```
$ git diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3a5fead..aed612b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -187,7 +187,7 @@ endif()
 
 # Libs
 set(hdrmerge_libs
-    "${LibRaw_r_LIBRARIES}"
+    "/opt/local/lib/libraw_r.dylib"
     "${exiv2_LIBRARIES}"
     "${ZLIB_LIBRARIES}"
 )
@@ -274,7 +274,7 @@ else()
     )
 endif()
 
-target_link_libraries(hdrmerge ${STRIP} ${hdrmerge_libs} Qt5::Widgets exiv2)
+target_link_libraries(hdrmerge ${STRIP} ${hdrmerge_libs} Qt5::Widgets /opt/local/lib/libexiv2.dylib)
 
 if(WIN32)
     # Compile a target without GUI, for the .com executable
diff --git a/src/ExifTransfer.cpp b/src/ExifTransfer.cpp
index cec5904..bc8f4f9 100644
--- a/src/ExifTransfer.cpp
+++ b/src/ExifTransfer.cpp
@@ -21,6 +21,7 @@
  */
 
 #include <exiv2/exiv2.hpp>
+#include <iostream>
 #include "ExifTransfer.hpp"
 #include "Log.hpp"
 using namespace hdrmerge;
```

Currently, `master` fails with:
```text
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find Exiv2: Found unsuitable version "..", but required is at
  least "0.12" (found /usr/lib64/libexiv2.so)
```

I confirm that @fanckush 's patch fixes compilation in `master` using exiv2-0.27.0. It produces:
```text
-- Found exiv2: /usr/lib64/libexiv2.so (found version "..")
```

How do we proceed - just apply @fanckush 's patch, or merge PR #174?

@Beep6581  

- building clean from master I get what you got:

``` zsh
CMake Error at /usr/local/Cellar/cmake/3.14.3/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find Exiv2: Found unsuitable version "..", but required is at
  least "0.12" (found /usr/local/Cellar/exiv2/0.27.0a/lib/libexiv2.dylib)
```

- Still master + change below (the patch), resulted is Cmake command succeeding ✔
> ... I switched to master and changed this:
> ```diff
> -find_package_handle_standard_args(Exiv2  REQUIRED_VARS  EXIV2_LIBRARY EXIV2_INCLUDE_DIR
> +find_package_handle_standard_args(exiv2  REQUIRED_VARS  EXIV2_LIBRARY EXIV2_INCLUDE_DIR
>                                           VERSION_VAR  EXIV2_VERSION)
> ```

However, `make install` fails for unrelated reason caused by `/usr/local/include/` not being used [macOS: /usr/local/include not a system include directory under CMake](https://gitlab.kitware.com/cmake/cmake/issues/19120)
the sepecfic error I got:
``` zsh
/usr/local/Cellar/exiv2/0.27.0a/include/exiv2/ssh.hpp:27:10: fatal error: 
      'libssh/libssh.h' file not found
#include <libssh/libssh.h>
         ^~~~~~~~~~~~~~~~~
```
in this case libssh can be found in `/usr/local/include/`

- After adding the include dir manually (for the sake of seeing if it will work or not):
``` diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9330805..cb14303 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,6 +59,7 @@ include_directories(
     "${EXIV2_INCLUDE_DIR}"
     "${ZLIB_INCLUDE_DIRS}"
     "${ALGLIB_INCLUDES}"
+    "/usr/local/include/"
 )
```
Now it builds successfully.

- On `cmake-exiv2` the `cmake` command works but `make install` fails

--- a/cmake/FindExiv2.cmake
+++ b/cmake/FindExiv2.cmake
@@ -1,62 +1,53 @@
-# - Try to find the Exiv2 library
-#
-#  EXIV2_MIN_VERSION - You can set this variable to the minimum version you need
-#                      before doing FIND_PACKAGE(Exiv2). The default is 0.12.
-#
+# - Try to find Exiv2
 # Once done this will define
 #
-#  EXIV2_FOUND - system has libexiv2
-#  EXIV2_INCLUDE_DIR - the libexiv2 include directory
-#  EXIV2_LIBRARIES - Link these to use libexiv2
-#  EXIV2_DEFINITIONS - Compiler switches required for using libexiv2
+#  EXIV2_FOUND - system has Exiv2
+#  EXIV2_INCLUDE_DIR - the Exiv2 include directory
+#  EXIV2_LIBRARIES - Link these to use Exiv2
+#  EXIV2_DEFINITIONS - Compiler switches required for using Exiv2
 #
-# The minimum required version of Exiv2 can be specified using the
-# standard syntax, e.g. find_package(Exiv2 0.17)
+#=============================================================================
+#  Copyright (c) 2019 Andres Schneider <asn@cryptomilk.org>
 #
-# For compatiblity, also the variable EXIV2_MIN_VERSION can be set to the minimum version
-# you need before doing FIND_PACKAGE(Exiv2). The default is 0.12.
-
-# Copyright (c) 2010, Alexander Neundorf, <neundorf@kde.org>
-# Copyright (c) 2008, Gilles Caulier, <caulier.gilles@gmail.com>
+#  Distributed under the OSI-approved BSD License (the "License");
+#  see accompanying file Copyright.txt for details.
 #
-# Redistribution and use is allowed according to the terms of the BSD license.
-# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+#  This software is distributed WITHOUT ANY WARRANTY; without even the
+#  implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the License for more information.
+#=============================================================================
+#
 
-# Support EXIV2_MIN_VERSION for compatibility:
-if(NOT Exiv2_FIND_VERSION)
-  set(Exiv2_FIND_VERSION "${EXIV2_MIN_VERSION}")
-endif(NOT Exiv2_FIND_VERSION)
+if (UNIX)
+  find_package(PkgConfig)
+  if (PKG_CONFIG_FOUND)
+    pkg_check_modules(_EXIV2 exiv2)
+  endif (PKG_CONFIG_FOUND)
+endif (UNIX)
 
-# the minimum version of exiv2 we require
-if(NOT Exiv2_FIND_VERSION)
-  set(Exiv2_FIND_VERSION "0.12")
-endif(NOT Exiv2_FIND_VERSION)
+find_path(EXIV2_INCLUDE_DIR
+    NAMES
+        exiv2/exif.hpp
+    PATHS
+        ${_EXIV2_INCLUDEDIR}
+)
 
+find_library(EXIV2_LIBRARY
+    NAMES
+        exiv2
+    PATHS
+        ${_EXIV2_LIBDIR}
+)
 
-if (NOT WIN32)
-   # use pkg-config to get the directories and then use these values
-   # in the FIND_PATH() and FIND_LIBRARY() calls
-   find_package(PkgConfig)
-   pkg_check_modules(PC_EXIV2 QUIET exiv2)
-   set(EXIV2_DEFINITIONS ${PC_EXIV2_CFLAGS_OTHER})
-endif (NOT WIN32)
-
-
-find_path(EXIV2_INCLUDE_DIR NAMES exiv2/exif.hpp
-          HINTS
-          ${PC_EXIV2_INCLUDEDIR}
-          ${PC_EXIV2_INCLUDE_DIRS}
-        )
-
-find_library(EXIV2_LIBRARY NAMES exiv2 libexiv2
-             HINTS
-             ${PC_EXIV2_LIBDIR}
-             ${PC_EXIV2_LIBRARY_DIRS}
-            )
-
+if (EXIV2_LIBRARY)
+    set(EXIV2_LIBRARIES
+        ${EXIV2_LIBRARIES}
+        ${EXIV2_LIBRARY}
+    )
+endif (EXIV2_LIBRARY)
 
 # Get the version number from exiv2/version.hpp and store it in the cache:
-if(EXIV2_INCLUDE_DIR  AND NOT  EXIV2_VERSION)
+if (EXIV2_INCLUDE_DIR AND NOT EXIV2_VERSION)
   file(READ ${EXIV2_INCLUDE_DIR}/exiv2/version.hpp EXIV2_VERSION_CONTENT)
   string(REGEX MATCH "#define EXIV2_MAJOR_VERSION +\\( *([0-9]+) *\\)"  _dummy "${EXIV2_VERSION_CONTENT}")
   set(EXIV2_VERSION_MAJOR "${CMAKE_MATCH_1}")
@@ -67,14 +58,18 @@ if(EXIV2_INCLUDE_DIR  AND NOT  EXIV2_VERSION)
   string(REGEX MATCH "#define EXIV2_PATCH_VERSION +\\( *([0-9]+) *\\)"  _dummy "${EXIV2_VERSION_CONTENT}")
   set(EXIV2_VERSION_PATCH "${CMAKE_MATCH_1}")
 
-  set(EXIV2_VERSION "${EXIV2_VERSION_MAJOR}.${EXIV2_VERSION_MINOR}.${EXIV2_VERSION_PATCH}" CACHE STRING "Version number of Exiv2" FORCE)
-endif(EXIV2_INCLUDE_DIR  AND NOT  EXIV2_VERSION)
+  set(EXIV2_VERSION "${EXIV2_VERSION_MAJOR}.${EXIV2_VERSION_MINOR}.${EXIV2_VERSION_PATCH}")
+endif (EXIV2_INCLUDE_DIR AND NOT EXIV2_VERSION)
 
-set(EXIV2_LIBRARIES "${EXIV2_LIBRARY}")
+if (EXIV2_VERSION)
+    set(EXIV2_FOUND TRUE)
+endif (EXIV2_VERSION)
 
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(Exiv2  REQUIRED_VARS  EXIV2_LIBRARY EXIV2_INCLUDE_DIR
-                                         VERSION_VAR  EXIV2_VERSION)
+find_package_handle_standard_args(Exiv2
+                                  FOUND_VAR EXIV2_FOUND
+                                  REQUIRED_VARS EXIV2_INCLUDE_DIR EXIV2_LIBRARY EXIV2_LIBRARIES
+                                  VERSION_VAR EXIV2_VERSION)
 
-mark_as_advanced(EXIV2_INCLUDE_DIR EXIV2_LIBRARY)
-
+# show the EXIV2_INCLUDE_DIR and EXIV2_LIBRARIES variables only in the advanced view
+mark_as_advanced(EXIV2_INCLUDE_DIR EXIV2_LIBRARY EXIV2_LIBRARIES)

